src/py/flwr/server/superlink/driver/serverappio_servicer.py

```diff
@@ -181,7 +185,40 @@ def PushMessages(
         self, request: PushMessagesRequest, context: grpc.ServicerContext
     ) -> PushMessagesResponse:
         """Push a set of Messages."""
-        return PushMessagesResponse(message_ids=[])
+        log(DEBUG, "ServerAppIoServicer.PushMessages")
+
+        # Init state
+        state: LinkState = self.state_factory.state()
+
+        # Abort if the run is not running
+        abort_if(
+            request.run_id,
+            [Status.PENDING, Status.STARTING, Status.FINISHED],
+            state,
+            context,
+        )
+
+        # Set pushed_at (timestamp in seconds)
+        pushed_at = time.time()
+
+        # Validate request and insert in State
+        _raise_if(len(request.messages_list) == 0, "`messages_list` must not be empty")
+        message_ids: list[Optional[UUID]] = []
+        for message_proto in request.messages_list:
```

Comment:
Should we use this to avoid high peak memory usage?
```suggestion
        while request.messages_list:
                message_proto = request.messages_list.pop(0)
```